pkg/openscap/openscap.go

```diff
@@ -191,7 +191,7 @@ func (s *defaultOSCAPScanner) oscapChroot(ctx context.Context, oscapArgs ...stri
 	return out, err
 }
 
-func (s *defaultOSCAPScanner) Scan(ctx context.Context, mountPath string, image *docker.Image) ([]iiapi.Result, interface{}, error) {
+func (s *defaultOSCAPScanner) Scan(ctx context.Context, mountPath string, image *docker.Image, filter iiapi.FilesFilter) ([]iiapi.Result, interface{}, error) {
```

Comment:
what's the purpose of having `filter` here? i recall the discussion here https://github.com/openshift/image-inspector/pull/72#discussion_r139773505

if `filter` is not used by all of the scanners, it shouldn't be a part of the interface. thoughts @pweil- ?

Reply:
> if filter is not used by all of the scanners, it shouldn't be a part of the interface. thoughts @pweil- ?

yes, I do not like it here.  I don't have a solution in mind but it feels like the wrong way to use the interface.

Reply:
@pweil- @ilackarms The filter is not used here. This is an optional parameter that can narrow the context of the review. If the scanner does not support such a restriction, then it can be ignored.

Reply:
IMO, there is nothing wrong with passing the filter to all scanners and allowing every scanner to decide whether to make use of it. Scan types that only operate in a holistic fashion might ignore it while file-wise scanner should use it.

Reply:
devil's advocate: out of the scanners we have today this works for 50% of them.  We've talked about scanning memory in the past too.  

Maybe what I'm getting at here is that rather than extending the scan interface signature endlessly we make a `ScannerOptions` object.  Then we're in the situation of supporting every possible option for supported scanners in that object which isn't really any different but makes the interface nice.

I think there also might be a pattern for `ScanOptions.Clam` and `ScanOptions.OpenSCAP` which allows more separation and validation on the scanner side.  But we still need a way to pass those back to the scanners from the cli....which then makes me start thinking we need to be able to pass these as a file.

tl;dr I want the interface to be short and the options to be scanner specific if possible

Reply:
@pweil- i would go a step further and argue that scanner-specific options should only be passed to 

imo, the purpose of using interfaces in the first place is to decouple the function that calls `scanner.Scan` from the concrete implementation of the scanner. if the caller needs to be aware of `ScanOptions.{Clam,OpenSCAP}`, doesn't that defeat the purpose of using an interface? i think only the section of code that's coupled with the concrete scanner (e.g. the concrete constructor) should be responsible for configuring scanner-specific options. 

in other words, i would suggest adding `ScanOptions` to the struct fields and keep them separate from the interface.

Reply:
The caller must populate the options at some point.  Whether that's hydrating a type from a file or having command line options.  The reason I like it in the `Scan` interface is that it would work nicely for a persistent scanner where `Scan` can be called multiple times with different options.  I think that's the mental line I draw between what would be `ScanOptions` and `ScannerOptions`.  

This is a good topic for arch but in the spirit of getting this in for the next milestone should we:

1.  add this to the interface
2.  add this to a ScanOptions on the interface
3.  add this to the Scanner instantiation itself

Realistically, we don't have external implementations of the Scanner interface so we're not backing ourselves into a corner either way at this point.  

Preferences or other opinions on this @simon3z @enoodle @mfojtik ?

Reply:
no one?  Well if that's the case then I think we should leave it as is until we agree on a solution.

Reply:
@pweil- i am fine with leaving as is